### DISCUSSION TITLE
Re: Overlaing text on hexgon tile map board...

### DISCUSSION THREAD

**Discussion by snaidamast** [MonoGame/MonoGame#8588] [Graphics]
Hello...

I have a hexagon map board that is being successfully drawn from the export of the corresponding Tile Map Editor files (TMX\TSX).  The image of this map board is shown with this post.

I would like to add tile positional text over each of the tiles.  For example, "0,0", would be the text for the tile/hexagon in the upper left-hand corner of the map.

I know that I can use the SpriteBatch.DrawString to write out such text by making it the second layer that is written when actual tile image is drawn using the SpriteBatch.Draw method.

What I would like to know is how I could implement a process that would allow a user to remove these text captions upon the selection of such an option...

Thank you...
![SampleHexagonMap_8x6](https://github.com/user-attachments/assets/58c91f7e-8b08-4910-8998-34439635a410)

**Comment by KelsamGames** [comment]
Maybe I'm misunderstanding something, but is this not as simple as having a boolean and only drawing the labels if it is true?

**Reply by snaidamast** [reply]
No it isn't....

What I am looking to do is draw the text-labels with my SpriteBatch, which will initially display them over each tile on the map board.

However, I want the option to erase them at any point.  My question is how do I erase or remove the labels after they have been displayed without having to redraw the entire map board again?

**Reply by KelsamGames** [reply]
Reading between the lines since I obviously can't see your code and don't know how your game works, therefore assuming you're not drawing this map every frame but instead have drawn it to an RT or some such that you're reusing: then just draw your labels to a separate RT from the one you've drawn your map to and then in your draw loop only draw the labels RT after drawing the map RT if the setting is enabled.

**Reply by snaidamast** [reply]
So drawing a secondary RenderTarget will overlay the first one?  If so, that was something I have thought about but was not sure if it work...

**Reply by KelsamGames** [reply]
Yeah, you can use Color.Transparent (or Color.TransparentBlack if you have specific shader issues or some such) to clear it if necessary. You need to make sure your BlendState and such is setup right for SpriteBatch.

**Reply by snaidamast** [reply]
I already have a transparent hexagon tile that I can use to draw the entire map, but to a second RenderTarget2D.  So a secondary map-board would only have transparent hexagon tiles with only a black outline in each.  I can then set each tile to have a text-caption as I already have calculated the pixel coordinates for each caption based on each tile's position in the map.

Is this what you are suggesting?  If so, that should be relatively straight forward to implement.

Thank you for all your help so far...   :-)
